docs/guides/database.rst

```diff
@@ -193,9 +215,21 @@ $type
 $subtype
 	Entity subtype, e.g. ``'blog'`` or ``'page'``.
 
-By convention entities that have an uploaded avatar or icon will have the ``icontime`` property
-assigned. This means that you can use ``$entity->icontime`` to check if an icon exists for the given
-entity.
+Icons are assigned a timestamp when they're created. You can use ``$object->getIconTs($size)`` to 
+get the timestamp if you need to track changes to the icon.
+
+To check if an icon is set, use ``$object->hasIcon($size)``.
+
+Icon methods support passing an icon type if an entity has more than one icon. For example, a user
+might have an avatar and a cover photo icon. You would pass ``'cover_photo'`` as the icon type:
+
+.. code:: php
+	$object->setIconFromUploadedFile('uploaded_photo', 'cover_photo');
+
+	$object->getIconUrl([
+		'size' => 'medium',
+		'type' => 'cover_photo'
```

Comment:
Agreed that "large" isn't so large anymore. I was thinking to extend the API to something like:

```php
[
   'icon' => [
       'large' => ['width' => 325, 'height' => 200, 'upscale' => true, 'downscale' => false]
       ...
    ],
    'cover_photo' => [
       'large' => ['width' => 900, 'height' => 300, 'upscale' => true, 'downscale' => 
    ]
]
```

Other options might also be useful: min_width, min_height...

Core has to have a way to know which sizes to serve / cache. The interface can be asking themes to register sizes.